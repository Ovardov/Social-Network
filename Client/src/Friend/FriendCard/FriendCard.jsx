import React from 'react';
import { Link } from 'react-router-dom';
import Avatar from '../../Avatar/Avatar';
import styles from './friend-card.module.scss';


function PostCard({ name, profilePicture, livesIn, id }) {

    return (
        <Link to={`/profile/${id}`}>
            <section className={styles.container}>
                <div className={styles['image-container']}>
                    <Avatar name={name} profilePicture={profilePicture}></Avatar>

                    <div className={styles['message-container']}>
                        <i class="fas fa-user-circle message-icon"></i>
                    </div>
                </div>

                <h5>{name}</h5>
                <p>Lives in {livesIn}</p>
            </section>
        </Link>
    )

}

export default PostCard;