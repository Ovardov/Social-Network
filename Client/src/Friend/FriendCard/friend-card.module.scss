.container {
    padding: 2em;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    position: relative;
    background: #FFFFFF;
    color: #000000;
    border: 1px solid #F4F4F4;
    border-radius: 0.3em;
    cursor: pointer;


    &:hover:after {
        display: block;
        content: '';
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        box-shadow: 0 0 0.3em 0 rgba(0, 0, 0, 0.5);
        border-radius: 0.3em;
    }

    & .image-container {
        display: flex;
        position: relative;
        justify-content: center;
        align-items: center;
        width: 6em;
        height: 6em;
        border: 1px solid #ffffff;
        border-radius: 50%;
        margin-bottom: 0.5em;

        & > div:first-child {
            width: 5em;
            height: 5em;
        }
    }

    &:hover > .image-container {
        border-color: var(--color-blue-main);
    }

    &:hover .message-container {
        display: flex;
    }

    & .message-container {
        display: none;
        justify-content: center;
        align-items: center;
        position: absolute;
        bottom: 0;
        right: 0.2em;
        z-index: 10;
        width: 2em;
        height: 2em;
        background: var(--color-blue-main);
        border-radius: 50%;

        & > i {
            color: #FFFFFF;
        }
        
    }

    & h5 {
        margin-bottom: 0;
    }

    & p {
        padding-bottom: 0;
        margin: 0;
        font-size: 0.9rem;
        color: var(--color-grey-main);
    }   
}