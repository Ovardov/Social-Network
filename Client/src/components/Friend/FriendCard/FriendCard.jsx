import React from 'react';
import { Link } from 'react-router-dom';
import Avatar from '../../Avatar/Avatar';
import styles from './friend-card.module.scss';


function FriendCard({ name, profilePicture, home, username, lastFriends }) {
    return (
        <Link to={`/profile/${username}`}>
            <section className={styles.container}>
                <div className={styles['image-container']}>
                    <Avatar name={name} profilePicture={profilePicture}></Avatar>

                    {lastFriends !== true && (
                        <div className={styles['message-container']}>
                            <i className="fas fa-user-circle message-icon"></i>
                        </div>
                    )}
                </div>

                <div className={styles.info}>
                    {lastFriends !== true && <h5 className={home ? '' : styles.scale}>{name}</h5>}

                    {lastFriends !== true && home && <p>Lives in {home}</p>}
                </div>
            </section>
        </Link>
    )

}

export default FriendCard;