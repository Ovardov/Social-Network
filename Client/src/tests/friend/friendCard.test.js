import React from 'react'
import renderer from 'react-test-renderer';
import FriendCard from '../../components/Friend/FriendCard/FriendCard';

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('FriendCard component', () => {
    test('should render correcly with given data when is not in last friends', () => {
        const component = renderer.create(
            <FriendCard
                username="Ovardov"
                name="Aleksandar Ovardov"
                profilePicture="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
                home="Plovdiv"
                lastFriends={false}/>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should render correcly with given data when is last friends', () => {
        const component = renderer.create(
            <FriendCard
                username="Ovardov"
                name="Aleksandar Ovardov"
                profilePicture="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
                home="Plovdiv"
                lastFriends={true}/>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should render correcly with given data without home', () => {
        const component = renderer.create(
            <FriendCard
                username="Ovardov"
                name="Aleksandar Ovardov"
                profilePicture="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
                lastFriends={false}/>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
})
