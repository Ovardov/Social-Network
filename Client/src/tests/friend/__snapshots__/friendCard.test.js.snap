// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendCard component should render correcly with given data when is last friends 1`] = `
<Link
  to="/profile/Ovardov"
>
  <section
    className="container"
  >
    <div
      className="image-container"
    >
      <span
        className="container"
        onClick={[Function]}
      >
        <Link
          to="/profile/undefined"
        >
          <img
            alt="Aleksandar Ovardov"
            className="avatar-image"
            src="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
          />
        </Link>
      </span>
    </div>
  </section>
</Link>
`;

exports[`FriendCard component should render correcly with given data when is not in last friends 1`] = `
<Link
  to="/profile/Ovardov"
>
  <section
    className="container"
  >
    <div
      className="image-container"
    >
      <span
        className="container"
        onClick={[Function]}
      >
        <Link
          to="/profile/undefined"
        >
          <img
            alt="Aleksandar Ovardov"
            className="avatar-image"
            src="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
          />
        </Link>
      </span>
      <div
        className="message-container"
      >
        <i
          className="fas fa-user-circle message-icon"
        />
      </div>
    </div>
    <h5>
      Aleksandar Ovardov
    </h5>
    <p>
      Lives in 
      Plovdiv
    </p>
  </section>
</Link>
`;

exports[`FriendCard component should render correcly with given data without home 1`] = `
<Link
  to="/profile/Ovardov"
>
  <section
    className="container"
  >
    <div
      className="image-container"
    >
      <span
        className="container"
        onClick={[Function]}
      >
        <Link
          to="/profile/undefined"
        >
          <img
            alt="Aleksandar Ovardov"
            className="avatar-image"
            src="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
          />
        </Link>
      </span>
      <div
        className="message-container"
      >
        <i
          className="fas fa-user-circle message-icon"
        />
      </div>
    </div>
    <h5>
      Aleksandar Ovardov
    </h5>
  </section>
</Link>
`;
